### PR TITLE
v1.11.0

### PR DESCRIPTION
**Pull request by TheTrunk**
Adjustments and improvements of Home Frontend
Dashboard section is now available on Flux
Preparations for Flux fork

This release has to be enforced as of new api and some message structures changes (that is subjected to change again later on)

### CODE REVIEW

File: ZelBack/src/routes.js [hunk 1]
```diff
@@ -1,912 +1,1770 @@
...
-  //   explorerService.getAllZelNodeTransactions(req, res);
-  // });
-  // filter can be IP, address, collateralHash.
-  app.get('/explorer/zelnodetxs/:filter?', cache('30 seconds'), (req, res) => {
-    explorerService.getFilteredZelNodeTxs(req, res);
-  });
-  app.get('/explorer/utxo/:address?', cache('30 seconds'), (req, res) => {
-    explorerService.getAddressUtxos(req, res);
-  });
-  app.get('/explorer/transactions/:address?', cache('30 seconds'), (req, res) => {
```
Comment (Cabecinha84) [inline ZelBack/src/routes.js:311]: Wrongly deleted?

File: ZelBack/src/routes.js [hunk 2]
```diff
@@ -1,912 +1,1770 @@
...
+    fluxService.restartBenchmark(req, res);
   });
   app.get('/zelbench/restartnodebenchmarks', (req, res) => {
-    zelbenchService.restartNodeBenchmarks(req, res);
-  });
-
-  app.get('/explorer/reindex/:reindexapps?', (req, res) => {
-    explorerService.reindexExplorer(req, res);
-  });
-  app.get('/explorer/restart', (req, res) => {
```
Comment (Cabecinha84) [inline ZelBack/src/routes.js:695]: wrongly deleted?

File: ZelBack/src/routes.js (hunk 1 above)
Comment (TheTrunk) [reply ZelBack/src/routes.js:311]: its still there but more down. 
APIs are now 'duplicatte' as of different path structure

File: ZelBack/src/routes.js (hunk 2 above)
Comment (TheTrunk) [reply ZelBack/src/routes.js:695]: its still there but more down.
APIs are now 'duplicatte' as of different path structure

File: ZelBack/src/routes.js [hunk 3]
```diff
@@ -1,912 +1,1770 @@
...
+  });
+  app.get('/apps/fluxshare/sharedfiles', (req, res) => {
+    appsService.fluxShareGetSharedFiles(req, res);
+  });
+  app.get('/apps/fluxshare/rename/:oldpath?/:newname?', (req, res) => {
+    appsService.fluxShareRename(req, res);
+  });
+  app.get('/apps/fluxshare/downloadfolder/:folder?', (req, res) => {
+    appsService.fluxShareDownloadFolder(req, res);
   });
```
Comment (Cabecinha84) [inline ZelBack/src/routes.js:1769]: Ok... Seems like they were just moved...

File: ZelBack/src/services/appsService.js [hunk 4]
```diff
@@ -1490,21 +1535,21 @@ async function removeZelAppLocally(zelapp, res, force = false, endResponse = tru
...
+      status: 'Flux App image operations done',
     };
     log.info(imageStatus2);
     if (res) {
       res.write(serviceHelper.ensureString(imageStatus2));
     }
 
     const portStatus = {
-      status: 'Denying ZelApp port...',
+      status: 'Denying Flux Appport...',
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:1546]: typo Flux App port

File: ZelBack/src/services/appsService.js [hunk 5]
```diff
@@ -1709,74 +1754,74 @@ async function removeZelAppLocally(zelapp, res, force = false, endResponse = tru
...
+    status: 'Flux App stopped',
   };
   log.info(stopStatus2);
   if (res) {
     res.write(serviceHelper.ensureString(stopStatus2));
   }
 
   const removeStatus = {
-    status: 'Removing ZelApp container...',
+    status: 'Removing Flux Appcontainer...',
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:1800]: typo App container...

File: ZelBack/src/services/appsService.js [hunk 6]
```diff
@@ -1709,74 +1754,74 @@ async function removeZelAppLocally(zelapp, res, force = false, endResponse = tru
...
+    status: 'Flux App container removed',
   };
   log.info(removeStatus2);
   if (res) {
     res.write(serviceHelper.ensureString(removeStatus2));
   }
 
   const imageStatus = {
-    status: 'Removing ZelApp image...',
+    status: 'Removing Flux Appimage...',
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:1818]: typo App Image...

File: ZelBack/src/services/appsService.js [hunk 7]
```diff
@@ -1788,21 +1833,21 @@ async function softRemoveZelAppLocally(zelapp, res) {
...
+    status: 'Flux App image operations done',
   };
   log.info(imageStatus2);
   if (res) {
     res.write(serviceHelper.ensureString(imageStatus2));
   }
 
   const portStatus = {
-    status: 'Denying ZelApp port...',
+    status: 'Denying Flux Appport...',
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:1844]: Typo App port

File: ZelBack/src/services/appsService.js [hunk 8]
```diff
@@ -1852,11 +1897,11 @@ async function removeZelAppLocallyApi(req, res) {
       force = serviceHelper.ensureBoolean(force);
 
       if (!appname) {
-        throw new Error('No ZelApp specified');
+        throw new Error('No Flux Appspecified');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:1900]: Typo App specified

File: ZelBack/src/services/appsService.js [hunk 9]
```diff
@@ -2027,12 +2072,12 @@ async function registerZelAppLocally(zelAppSpecifications, res) {
             res.write(serviceHelper.ensureString(errorResponse));
           }
 
-          const removeStatus = serviceHelper.createErrorMessage('Error in volume assigning occured. Initiating ZelApp removal');
+          const removeStatus = serviceHelper.createErrorMessage('Error in volume assigning occured. Initiating Flux Appremoval');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2075]: Typo App removal

File: ZelBack/src/services/appsService.js [hunk 10]
```diff
@@ -2120,20 +2165,20 @@ async function registerZelAppLocally(zelAppSpecifications, res) {
           if (res) {
             res.write(serviceHelper.ensureString(errorResponse));
           }
-          const removeStatus = serviceHelper.createErrorMessage('Error occured. Initiating ZelApp removal');
+          const removeStatus = serviceHelper.createErrorMessage('Error occured. Initiating Flux Appremoval');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2168]: typo App removal

File: ZelBack/src/services/appsService.js [hunk 11]
```diff
@@ -2209,35 +2254,35 @@ async function softRegisterZelAppLocally(zelAppSpecifications, res) {
...
-    if (zelappResult) {
-      throw new Error('ZelApp already installed');
+    const appResult = await serviceHelper.findOneInDatabase(appsDatabase, localAppsInformation, appsQuery, appsProjection);
+    if (appResult) {
+      throw new Error('Flux App already installed');
     }
 
     const checkParameters = {
-      status: 'Checking ZelApp requirements...',
+      status: 'Checking Flux Apprequirements...',
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2263]: typo App requirements...

File: ZelBack/src/services/appsService.js [hunk 12]
```diff
@@ -2566,32 +2611,32 @@ async function verifyAppHash(message) {
   const messToHash = message.type + message.version + JSON.stringify(message.zelAppSpecifications) + message.timestamp + message.signature;
   const messageHASH = await messageHash(messToHash);
   if (messageHASH !== message.hash) {
-    throw new Error('Invalid ZelApp hash received!');
+    throw new Error('Invalid Flux Apphash received!');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2614]: typo App hash

File: ZelBack/src/services/appsService.js [hunk 13]
```diff
@@ -2566,32 +2611,32 @@ async function verifyAppHash(message) {
...
   }
   return true;
 }
 
-async function verifyZelAppMessageSignature(type, version, zelAppSpec, timestamp, signature) {
-  if (typeof zelAppSpec !== 'object' && typeof timestamp !== 'number' && typeof signature !== 'string' && typeof version !== 'number' && typeof type !== 'string') {
-    throw new Error('Invalid ZelApp message specifications');
+async function verifyAppMessageSignature(type, version, appSpec, timestamp, signature) {
+  if (typeof appSpec !== 'object' && typeof timestamp !== 'number' && typeof signature !== 'string' && typeof version !== 'number' && typeof type !== 'string') {
+    throw new Error('Invalid Flux Appmessage specifications');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2621]: typo App message

File: ZelBack/src/services/appsService.js [hunk 14]
```diff
@@ -2778,7 +2826,7 @@ async function storeZelAppTemporaryMessage(message, furtherVerification = false)
   * @param signature string
   */
   if (typeof message !== 'object' && typeof message.type !== 'string' && typeof message.version !== 'number' && typeof message.zelAppSpecifications !== 'object' && typeof message.signature !== 'string' && typeof message.timestamp !== 'number' && typeof message.hash !== 'string') {
-    return new Error('Invalid ZelApp message for storing');
+    return new Error('Invalid Flux Appmessage for storing');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2829]: typo App message

File: ZelBack/src/services/appsService.js [hunk 15]
```diff
@@ -2788,51 +2836,51 @@ async function storeZelAppTemporaryMessage(message, furtherVerification = false)
...
       }
-      const { owner } = zelappInfo;
-      // here signature is checked against PREVIOUS zelapp owner
-      await verifyZelAppMessageUpdateSignature(message.type, message.version, message.zelAppSpecifications, message.timestamp, message.signature, owner);
+      const { owner } = appInfo;
+      // here signature is checked against PREVIOUS app owner
+      await verifyAppMessageUpdateSignature(message.type, message.version, message.zelAppSpecifications, message.timestamp, message.signature, owner);
     } else {
-      throw new Error('Invalid ZelApp message received');
+      throw new Error('Invalid Flux Appmessage received');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2872]: typo App message

File: ZelBack/src/services/appsService.js [hunk 16]
```diff
@@ -2863,7 +2911,7 @@ async function storeZelAppRunningMessage(message) {
   * @param ip string
   */
   if (typeof message !== 'object' && typeof message.type !== 'string' && typeof message.version !== 'number' && typeof message.broadcastedAt !== 'number' && typeof message.hash !== 'string' && typeof message.name !== 'string' && typeof message.ip !== 'string') {
-    return new Error('Invalid ZelApp Running message for storing');
+    return new Error('Invalid Flux AppRunning message for storing');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:2914]: Typo App Running

File: ZelBack/src/services/appsService.js [hunk 17]
```diff
@@ -2925,51 +2973,51 @@ async function registerZelAppGlobalyApi(req, res) {
...
+      let { containerData } = appSpecification;
+      let { cpu } = appSpecification;
+      let { ram } = appSpecification;
+      let { hdd } = appSpecification;
+      const { tiered } = appSpecification;
 
       // check if signature of received data is correct
       if (!version || !name || !description || !repotag || !owner || !port || !enviromentParameters || !commands || !containerPort || !containerData || !cpu || !ram || !hdd) {
-        throw new Error('Missing ZelApp specification parameter');
+        throw new Error('Missing Flux Appspecification parameter');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:3020]: typo App specification

File: ZelBack/src/services/appsService.js [hunk 18]
```diff
@@ -2985,7 +3033,7 @@ async function registerZelAppGlobalyApi(req, res) {
           envParamsCorrected.push(param);
         });
       } else {
-        throw new Error('Enviromental parameters for ZelApp are invalid');
+        throw new Error('Enviromental parameters for Flux Appare invalid');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:3036]: typo Flux App are

File: ZelBack/src/services/appsService.js [hunk 19]
```diff
@@ -3126,51 +3174,51 @@ async function updateZelAppGlobalyApi(req, res) {
...
+      let { containerData } = appSpecification;
+      let { cpu } = appSpecification;
+      let { ram } = appSpecification;
+      let { hdd } = appSpecification;
+      const { tiered } = appSpecification;
 
       // check if signature of received data is correct
       if (!version || !name || !description || !repotag || !owner || !port || !enviromentParameters || !commands || !containerPort || !containerData || !cpu || !ram || !hdd) {
-        throw new Error('Missing ZelApp specification parameter');
+        throw new Error('Missing Flux Appspecification parameter');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:3221]: typo App specification

File: ZelBack/src/services/appsService.js [hunk 20]
```diff
@@ -3186,7 +3234,7 @@ async function updateZelAppGlobalyApi(req, res) {
           envParamsCorrected.push(param);
         });
       } else {
-        throw new Error('Enviromental parameters for ZelApp are invalid');
+        throw new Error('Enviromental parameters for Flux Appare invalid');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:3237]: typo Flux App are

File: ZelBack/src/services/appsService.js [hunk 21]
```diff
@@ -4391,17 +4439,17 @@ async function checkAndNotifyPeersOfRunningApps() {
...
     if (installedAppsRes.status !== 'success') {
       throw new Error('Failed to get installed Apps');
     }
-    const runningAppsRes = await listRunningZelApps();
+    const runningAppsRes = await listRunningApps();
     if (runningAppsRes.status !== 'success') {
       throw new Error('Unable to check running Apps');
     }
-    const installedApps = installedAppsRes.data;
+    const installedA = installedAppsRes.data;
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:4450]: why the const name changed? better perception like it was before.

File: ZelBack/src/services/appsService.js [hunk 22]
```diff
@@ -4496,35 +4544,35 @@ async function expireGlobalApplications() {
...
 
     // get list of locally installed apps.
-    const installedAppsRes = await installedZelApps();
+    const installedAppsRes = await installedApps();
     if (installedAppsRes.status !== 'success') {
       throw new Error('Failed to get installed Apps');
     }
-    const installedApps = installedAppsRes.data;
-    const appsToRemove = installedApps.filter((app) => appNamesToExpire.includes(app.name));
+    const installedA = installedAppsRes.data;
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:4568]: previous const name was better.

File: ZelBack/src/services/appsService.js [hunk 23]
```diff
@@ -4740,19 +4788,19 @@ async function getAppPrice(req, res) {
...
+      let { name } = appSpecification;
+      let { cpu } = appSpecification;
+      let { ram } = appSpecification;
+      let { hdd } = appSpecification;
+      const { tiered } = appSpecification;
 
       // check if signature of received data is correct
       if (!name || !cpu || !ram || !hdd) {
-        throw new Error('Missing ZelApp HW specification parameter');
+        throw new Error('Missing Flux AppHW specification parameter');
```
Comment (Cabecinha84) [inline ZelBack/src/services/appsService.js:4803]: typo App HW

**Comment by Cabecinha84** [comment]
None of rcp calls of Benchmark is working

**Comment by Cabecinha84** [comment]
On Apps, shows kadenachainweb running, but Visit button doesn't work.

**Comment by Cabecinha84** [comment]
On Apps, shows kadenachainweb running, but none app on installed.

**Comment by Cabecinha84** [comment]
Flux Admin - Menage Daemon -> Don't show up anything.

**Comment by Cabecinha84** [comment]
When pressing Dashboard, while it was loading overview, got a network error.

**Review by Cabecinha84 — APPROVED** [review]
Ack

**Pull request merged**